### DISCUSSION TITLE
[Package Request] WSL 2

### DISCUSSION THREAD

**Discussion by kawofong** [microsoft/winget-pkgs#3669] [General] [answered]
# Package Requested

Is it possible to add WSL 2 to winget?

https://docs.microsoft.com/en-us/windows/wsl/install-win10

**Comment by denelon** [comment]
WSL isn't a "package" per se' It's more of a Windows Feature. We're looking to enable it.

**Reply by ghost** [reply]
@denelon I think he meant a package for the [Linux kernel package for upgrading from WSL 1 to WSL 2](https://docs.microsoft.com/en-us/windows/wsl/install-win10#step-4---download-the-linux-kernel-update-package)

**Comment by jedieaston** [comment]
The WSL team has said that there will be a one-shot command to enable and update WSL in the 21H1 update to Windows 10 (https://devblogs.microsoft.com/commandline/the-windows-subsystem-for-linux-build-2020-summary/#wsl-install). I think that's the reason nobody put the manifest in repo (since presumably winget will be released along with this command).

**Comment by vedantmgoyal9** [comment]
WSL will now be shipped through the MS store, I think.

**Comment by jedieaston** [comment, marked as answer]
`wsl --install` is now available in Windows 10 2004 and above. I think that's a better solution rather than winget having a special case for it.

**Reply by denelon** [reply]
We're looking to enable the Windows Feature support in WinGet 1.7. With that we will be able to have the WSL "installer" take a dependency on the Windows Feature for Virtualization. Any Linux distributions in the community repository can then "depend" on WSL, and WinGet should be able to install things gracefully.

**Comment by mhekel** [comment]
Nothing worse than when people really dont think about why a person is posing a certain question!!!! yes there are some who dont know how to enable it - i'm, not one of them but my queastion is the same. wsl was accidently deleted - there is no enabling it or typing wsl --install.. can't get it from the store because IT'S NOT THERE... wsl2 is NOT in the store by itself. Yes wsl2 is a feature more so than a package, dism doesn't do a damn thing to repair windows, and windows hangs up on repairing. (please don't suggest doing by usb, that isn't even an option to bootup by usb) - so can I use winget to get wsl back or am i sol?
11th Gen Intel(R) Core(TM) i5-11400H @ 2.70GHz   2.69 GHz 
Windows 11 Pro
Version	23H2
Installed on	‎1/‎7/‎2024
OS build	22631.3447
Experience	Windows Feature Experience Pack 1000.22688.1000.0

**Reply by denelon** [reply]
We've done the work to support enabling the Windows Feature dependency, and we will be adding a manifest for WSL so it can also be called as a dependency for Linux distros in the repository. I've been working with Craig on the WSL team to get to this point. I'm expecting it will be added in the coming days.

**Comment by mhekel** [comment]
For those who didn't realize that in asking this question that ``` wsl --install ``` wasn't an option, here is the actual solution https://github.com/microsoft/WSL/issues/11483#issuecomment-2067419606
